modules/ding_adgangsplatformen/ding_adgangsplatformen.module

```diff
@@ -0,0 +1,421 @@
+    unset($_SESSION['oauth2state']);
+    unset($_SESSION['oauth2ajax']);
+
+    drupal_set_message(t('Invalid state detected. Login failed'));
+    _ding_adgangsplatformen_redirect_user();
+  }
+  else {
+    try {
+      // Try to get an access token using the authorization code grant.
+      $provider = ding_adgangsplatformen_get_provider();
+      $access_token = $provider->getAccessToken('authorization_code', ['code' => $_GET['code']]);
+
+      // Store access token for later usage.
+      $_SESSION['oauth2token'] = $access_token->getToken();
+
+      // Get information about the user. This is a mis-use of the
+      // getResourceOwnerDetailsUrl configuration.
+      $request = $provider->getAuthenticatedRequest('POST', $provider->getResourceOwnerDetailsUrl($access_token), $access_token);
+      $response = $provider->getResponse($request);
+      $user_info = $response->getBody()->getContents();
+      $user_info = drupal_json_decode($user_info);
+
+      if (module_exists('ding_registration') && ding_registration_is_registration_request()) {
+        // This is an self registration request, so we store the information in
+        // the current session. So it can be used to create the user in the
+        // library system in the user creation flow in ding_registration.
+        // The redirect below will start the process.
+        ding_registration_set_user_information($user_info);
+      }
+      else {
+        // Login using the provider.
+        if (!_ding_adgangsplatformen_provider_login($user_info)) {
+          ding_adgangsplatformen_logout();
+          _ding_adgangsplatformen_redirect_user('<front>');
+        }
+      }
+
+      // Redirect user base on values set in the current session.
+      _ding_adgangsplatformen_redirect_user();
+    }
+    catch (\League\OAuth2\Client\Provider\Exception\IdentityProviderException $e) {
+      // Failed to get the access token or user details.
+      watchdog_exception('ding_adgangsplatformen', $e);
+
+      drupal_set_message(t('Failed to login and get user details for the user.'), 'warning');
+
+      // Clean up session variables to prevent redirects loop.
+      unset($_SESSION['oauth2token']);
+      unset($_SESSION['oauth2state']);
+
+      // Redirect user back.
+      _ding_adgangsplatformen_redirect_user();
+    }
+  }
+}
+
+/**
+ * Get oAuth2 provider.
+ *
+ * @return \League\OAuth2\Client\Provider\GenericProvider
+ *   Provider with basic configuration.
+ *
+ * @throws Exception
+ *   If required libraries are not loaded.
+ */
+function ding_adgangsplatformen_get_provider() {
+  // The oAuth2 library also uses guzzle.
+  foreach (array('random_compat', 'oauth2-client') as $lib) {
+    // Checking for FALSE, as it's actually a count of how many files was
+    // included, and none is included for our libraries.
+    if (!($library = libraries_load($lib)) || $library['loaded'] === FALSE) {
+      throw new Exception($lib . ' library not installed');
```

Comment:
The patch seams to be more that 5 years old and they do not agree on the solution in the issue.

Maybe some one should make the checks for installed libraries in ding_libs.

So removed the check here for now as it will throw an error is the libs are not installed.